### DISCUSSION TITLE
Smart Start process

### DISCUSSION THREAD

**Discussion by georgeinva2004** [zwave-js/ZWaveJS.NET#17] [General]
This is where my ignorance of ZWave is going to show.  I apologize in advance if this is a dumb question.  

I successfully integrated the new version of ZWaveJS.NET into my app.  I can capture the QR Code on a ZWave device successfully.  Unfortunately, when I pass the code to the ProvisionSmartStartNode method, I get an error back saying that the QR Code needs to begin with a "90".  The QR Code on the device (Sengled light bulb) seems to match the serial number on the box.  Is there a step I'm missing?

**Comment by marcus-j-davies** [comment]
Hi @georgeinva2004,

This error is from the driver.
Smart Start QR Codes don't actually have the serial  number of the device in them - they contain a large amount of information, but not the serial number to my knowledge.

a Smart Start QR Code will _usually_ have a 5 digit number printed along with it.
below is one from one of my devices - I have changed the Security data inside of it (DSK) - for obvious reasons 😅 

Use the below, and you should see it will be accepted (if I haven't messed it up)
`900143493001111112222233333444445555566666777770089900100179303072022000881000020000700256`

The 5 digit pin (that may also be printed) is the first 5 digit segment. of the 5x8 digit DSK (highlighted below)

900143493001`11111`2222233333444445555566666777770089900100179303072022000881000020000700256.

Its very likely the QR code you're scanning is not a smart Start on, S2 support is needed on the device for Smart Start

![image](https://user-images.githubusercontent.com/55892693/172200365-ba4e699a-6ab8-41ae-9009-cca070ebd9fe.png)

Once a QR has been accepted   - the device will auto pair the next time it is powered on (and not already attached to a network of course)

**Comment by marcus-j-davies** [comment]
> I apologize in advance if this is a dumb question.

and don't apologize 😉

**Comment by georgeinva2004** [comment]
A couple of things.  Came across this post from Sengled saying that the QR code on the box may have been misprinted, so I used the QR Code on the bulb itself.  https://support.sengled.com/hc/en-us/articles/360001982014-I-scanned-the-Element-box-QR-code-and-it-said-it-was-not-a-Sengled-product-What-do-I-do-

The QR Code on the bulb is this(X's added by me to obfuscate the actual code):  
![image](https://user-images.githubusercontent.com/5377265/172213817-624ba34f-9736-4dea-ae7f-1aa8af1584f6.png).  It kinda matches the code in the docs, but not exactly.  I don't see anything on the bulb that specifically says it's a "Smart Start compatible device".  I'm thinking that this isn't truly a Smart Start device, or I need to adjust the QR Code somehow...

**Reply by marcus-j-davies** [reply]
My Thought on it, is that it is not Smart Start capable 😞 - Or they have **really!!** messed up the QR code 😬 
The Smart Start QR codes are numerical only and should contain no special characters and of course start with 90.

A Smart Start QR example is below.
Code blocked numbers are changed for security - as this range is the encryption key (The DSK)
900147848132`1111122222333334444455555666667777788888`00101639508193022000838002570051400265

**Reply by georgeinva2004** [reply]
I think I figured it out (and the reason is embarrassing).  I took a look at the packaging on the light bulbs and discovered that they were Zigbee not ZWave.  So, a very rookie mistake on my part...

**Reply by marcus-j-davies** [reply]
They both begin with Z - so only a little mistake 😉
